framework/src/actions/BatchMeshGeneratorAction.C

```diff
@@ -0,0 +1,554 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// MOOSE includes
+#include "BatchMeshGeneratorAction.h"
+#include "ActionFactory.h"
+
+#include "MooseMesh.h"
+#include "MeshGenerator.h"
+#include "Factory.h"
+#include "MooseApp.h"
+
+#include "hit/hit.h"
+
+registerMooseAction("MooseApp", BatchMeshGeneratorAction, "add_mesh_generator");
+
+InputParameters
+BatchMeshGeneratorAction::validParams()
+{
+  InputParameters params = Action::validParams();
+
+  params.set<std::string>("type") = "BatchMeshGeneratorAction";
+
+  params.addClassDescription("Batch generate meshes using actions.");
+  params.addRequiredParam<std::string>("mesh_generator_name",
```

Comment:
```suggestion
  params.addRequiredParam<std::string>("mesh_generator_type",
```
or mest_generator_object_name